packages/bruno-common/src/interpolate/index.spec.ts

```diff
@@ -169,3 +169,131 @@ describe('interpolate - value edge cases', () => {
     expect(result).toBe(inputString);
   });
 });
+
+describe('interpolate - recursive', () => {
+  it('should replace placeholders with 1 level of recursion with values from the object', () => {
+    const inputString = '{{user.message}}';
+    const inputObject = {
+      'user.message': 'Hello, my name is {{user.name}} and I am {{user.age}} years old',
+      'user.name': 'Bruno',
+      user: {
+        age: 4
+      }
+    };
+
+    const result = interpolate(inputString, inputObject);
+
+    expect(result).toBe('Hello, my name is Bruno and I am 4 years old');
+  });
+
+  it('should replace placeholders with 2 level of recursion with values from the object', () => {
+    const inputString = '{{user.message}}';
+    const inputObject = {
+      'user.message': 'Hello, my name is {{user.name}} and I am {{user.age}} years old',
+      'user.name': 'Bruno {{user.lastName}}',
+      'user.lastName': 'Dog',
+      user: {
+        age: 4
+      }
+    };
+
+    const result = interpolate(inputString, inputObject);
+
+    expect(result).toBe('Hello, my name is Bruno Dog and I am 4 years old');
+  });
+
+  it('should replace placeholders with 3 level of recursion with values from the object', () => {
```

Comment:
There is only 2 level of recursion

Reply:
There is
{{user.message}}
{{user.full_name}}
And the full name has {{user.name}} and {{user.lastName}}

Do you can explain why there is only 2 levels of recursion?

Reply:
OK so if you count like this, the 2 tests above have 1 level of recursion more than in the title right ?

Reply:
you are right